.github/workflows/ubuntu-20.04.yml

```diff
@@ -332,9 +314,31 @@ jobs:
         docker stop mlos-${{ matrix.Configuration }}-build-ubuntu-${{ matrix.UbuntuVersion }}
         docker rm mlos-${{ matrix.Configuration }}-build-ubuntu-${{ matrix.UbuntuVersion }}
 
+  # Extra rules that only need to be run on a single Ubuntu version are injected here by gen-workflows.sh:
+
+# vim: set ft=yaml:
+#
+# NOTE: These rules are appended to the ubuntu-20.04.yml workflow by gen-workflows.sh
+# so that they only get applied once without a bunch of extra if clauses to maintain.
+# See Also: ubuntu.yml.tmpl for more details on how the docker usage works.
+
+  gen-workflows-check:
+    runs-on: ubuntu-latest
+    needs: [prep-vars]
+    if: needs.prep-vars.outputs.UbuntuVersionMatrix != '["SKIP_ME"]'
+    steps:
+    - uses: actions/checkout@v2
+    - name: Run the gen-workflows.sh script and check for changes that are not checked in
+      shell: bash
+      run: |
+        ./.github/workflows/gen-workflows.sh
+        git --no-pager diff --exit-code
```

Comment:
This is a new check to make sure that the workflow templating script is run and we don't find any differences.
This checks two things:
1. That not individual `ubuntu-{version}.yml` has been edited directly.
2. That the individual `ubuntu-{version}.yml` files that are checked in match what the script/templates say they should.